emanote/src/Emanote/View/Feed.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Emanote.View.Feed where
+
+import Data.Aeson.Optics (key, _String)
+import Data.Time (UTCTime)
+import Data.Time.Format
+import Emanote.Model (Model)
+import Emanote.Model.Note (Note (..))
+import Emanote.Model.Query (parseQuery, runQuery)
+import Emanote.Model.Title (toPlain)
+import Emanote.Route.ModelRoute
+import Emanote.Route.R (indexRoute)
+import Emanote.Route.SiteRoute
+import Optics.Operators ((^?))
+import Optics.Optic ((%))
+import Relude
+import Text.Atom.Feed qualified as Atom
+import Text.Atom.Feed.Export qualified as Export (textFeed)
+
+getNotes :: Text -> Model -> [Note]
+getNotes tagName model = case parseQuery ("tag:" <> tagName) of
+  Nothing -> []
+  Just query -> runQuery (LMLRoute_Md indexRoute) model query
+
+notesToFeed :: (Note -> Text) -> Atom.Date -> [Note] -> Atom.URI -> Text -> Atom.Feed
+notesToFeed noteUrl feedUpdated notes feedUrl title =
+  (Atom.nullFeed feedUrl feedTitle feedUpdated) {Atom.feedEntries}
+  where
+    feedTitle = Atom.TextString title
+    feedEntries = map toEntry notes
+    toEntry :: Note -> Atom.Entry
+    toEntry note = Atom.nullEntry (noteUrl note) noteTitle noteDate
+      where
+        noteDate = fromMaybe "1970-01-01" $ (_noteMeta note) ^? key "date" % _String
+        noteTitle = (Atom.TextString (toPlain $ _noteTitle note))
+
+renderFeed :: UTCTime -> Model -> Note -> LByteString
+renderFeed now model baseNote = encodeUtf8 $ fromMaybe "<error>bad feed</error>" mFeedTxt
+  where
+    mFeedTxt = do
+      feedSlug <- meta ^? key "slug" % _String
+      feedUrl <- meta ^? key "feed_url" % _String
+      feedName <- meta ^? key "feed_name" % _String
+      let noteUrl note =
```

Comment:
Instead of,

```yaml
generate_feed: true
feed_url: https://tristancacqueray.github.io
feed_name: Tristan's Zettelkasten
```

let's go with:

```yaml
feed:
  enable: true
  url: https://tristancacqueray.github.io
  title: Tristan's Zettelkasten
```

Actually, what is `url` doing here? Shouldn't it be in `index.yaml`? If you have multiple feeds, they all have a common base URL.

Reply:
Perhaps the title could default to the _noteTitle too?

Reply:
It doesn't seem like `index.yaml` already has a `url`, so I think we should fallback to the index only if the feed doesn't have one already. Could you please show how to get the `_noteMeta` of the `index.yaml` ?

Reply:
index.yaml can contain the defaults, which then gets merged with `.md`'s frontmatter YAML

This function automatically does the merge (using `SData.mergeAesons`):

https://github.com/srid/emanote/blob/84aaaac7384571576506377ccdd5182fc5662ac7/emanote/src/Emanote/Model/Meta.hs#L18-L21

https://github.com/search?q=repo%3Asrid%2Femanote%20lookupRouteMeta&type=code

Reply:
Alright thanks! So last commit pulls the url from either:

```
feed:
  url: X
```

Or from the index.yaml:

```
site:
  url: X
```